### PR TITLE
feat(atenlib): op(trunc)

### PR DESCRIPTION
**Pull request by justinchuby**
aten::trunc

```
<
  domain: "onnxscript.atenlib",
  opset_import: ["" : 18]
>
aten_trunc (self) => (return_val)
{
   tmp = Abs (self)
   integer_parts = Floor (tmp)
   tmp_0 = Constant <value = float tmp_0 {0}> ()
   tmp_0_cast = CastLike (tmp_0, self)
   is_negative = Less (self, tmp_0_cast)
   tmp_1 = Neg (integer_parts)
   return_val = Where (is_negative, tmp_1, integer_parts)
}
```

### CODE REVIEW

File: onnxscript/function_libs/torch_aten/ops/core.py [hunk 1]
```diff
@@ -5573,10 +5573,14 @@ def aten_true_divide(self: TensorType, other: TensorType) -> TensorType:
...
 
-def aten_trunc(self: TensorType) -> TensorType:
+@torch_op("aten::trunc")
+def aten_trunc(self: TFloatOrBFloat16) -> TFloatOrBFloat16:
     """trunc(Tensor self) -> Tensor"""
 
-    raise NotImplementedError()
+    # Reference https://github.com/onnx/onnx/issues/4588#issuecomment-1463970126
+    integer_parts = op.Floor(op.Abs(self))
+    is_negative = op.Less(self, 0.0)
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply onnxscript/function_libs/torch_aten/ops/core.py:5582]: ok!

[the remark this answers is not in the document]
Comment (justinchuby) [reply onnxscript/function_libs/torch_aten/ops/core.py:5582]: Looks like its here https://microsoft.github.io/onnx-script/tutorial/index.html#Automatic%20casts%20for%20constant%20values:~:text=Automatic%20casts%20for%20constant%20values

File: onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py [hunk 2]
```diff
@@ -389,6 +389,7 @@ def _where_input_wrangler(
     "tan": core_ops.aten_tan,
     "tanh": core_ops.aten_tanh,
     "topk": core_ops.aten_topk,
+    "trunc": core_ops.aten_trunc,
```
Comment (BowenBao) [inline onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py:392]: I wonder if there is anything we can do to improve the clarity on actual test cases. Seems hard to check without running it.

Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py:392]: Any suggestions? What kind of information will be useful?

**Pull request merged**